### PR TITLE
fix: atomic query builder actions

### PR DESCRIPTION
**Pull request by drdelambre**
Got some feedback on the "always select a measurement first" product change made in #4286. measurement selection should mirror bucket selection for a product perspective in that it should clear the query and start over again when you select a new one.

this involved some clean up on bucket selection as well. we used some useEffects to sniff on changes to the data.buckets property to see if the bucket had changed, and thus we should perform some action on the data. Now that we wanted that action to also be connected to a measurement, but only if there was a flag on, the useEffect got a bit... tangled. In order to make the code a bit more straight forward, the useEffect logic for clearing a query when a bucket gets selected got moved to it's own function on the context. once this proved a bit more forward, since we wanted a product change that mirrored this behavior, the technique was mirrored for measurement selection. this mechanism is a bit more rough because, for backwards compatibility reasons, we need to say that a measurement is the first card, and only if deemed 'compliant'.

### CODE REVIEW

File: src/flows/pipes/QueryBuilder/AddButton.tsx [hunk 1]
```diff
@@ -19,6 +20,14 @@ const AddButton: FC = () => {
     return null
   }
 
+  if (
+    isFlagEnabled('newQueryBuilder') &&
```
Comment (asalem1) [inline src/flows/pipes/QueryBuilder/AddButton.tsx:24]: it's like the old one, but newer

File: src/flows/pipes/QueryBuilder/BucketSelector.tsx [hunk 2]
```diff
@@ -67,7 +38,14 @@ const BucketSelector: FC = () => {
       </BuilderCard>
     )
   }
-
+  const _selectBucket = (bucket?: string) => {
+    if (bucket) {
+      event('Query Builder Bucket Selected', {
+        usedSearch: !!search.length ? 'yus' : 'nah',
```
Comment (asalem1) [inline src/flows/pipes/QueryBuilder/BucketSelector.tsx:44]:
```suggestion
        usedSearch: !!search.length,
``` 
I've got a sense of humor somewhere, but shouldn't this just be fine?

File: src/flows/pipes/QueryBuilder/CardList.tsx [hunk 3]
```diff
@@ -125,23 +131,27 @@ const Card: FC<Props> = ({idx}) => {
...
+      if (index === -1) {
+        event('Query Builder Value Selected')
+        selectMeasurement(val)
       } else {
-        _vals.push(val)
+        event('Query Builder Value Unselected')
+        selectMeasurement(null)
       }
+
+      return
```
Comment (asalem1) [inline src/flows/pipes/QueryBuilder/CardList.tsx:146]: confused on this early return. Shouldn't we update the values for this idx anyways to clear it out? Or is that handled elsewhere?

**Review by asalem1 — APPROVED** [review]
Some questions but nothing sus.

File: src/flows/pipes/QueryBuilder/AddButton.tsx (hunk 1 above)
Comment (drdelambre) [reply src/flows/pipes/QueryBuilder/AddButton.tsx:24]: wait for the `newerQueryBuilder` flag

File: src/flows/pipes/QueryBuilder/BucketSelector.tsx (hunk 2 above)
Comment (drdelambre) [reply src/flows/pipes/QueryBuilder/BucketSelector.tsx:44]: gotta be a string, and stop being boring

File: src/flows/pipes/QueryBuilder/CardList.tsx (hunk 3 above)
Comment (drdelambre) [reply src/flows/pipes/QueryBuilder/CardList.tsx:146]: yeah, so all of the stuff done below is moved out into the context function. but the context  function is only used for  the  _measurement card

Comment (asalem1) [reply src/flows/pipes/QueryBuilder/CardList.tsx:146]: Cool

**Pull request merged**